x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmsSettings.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.security.authc.jwt;
+
+import org.elasticsearch.common.settings.Setting;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Settings shared by all JWT realms.
+ */
+public class JwtRealmsSettings {
```

Comment:
I'd like us to rename this class, but it can be a followup PR.

I think we should avoid having classes that different only by an embedded `s`
- `JwtRealmSettings`
- `JwtRealmsSettings`

And likewise `JwtRealm` vs `JwtRealms`.

Can you explain your choice to rename `JwtAuthenticationTokenFactory` to `JwtRealms` during the evolution of this PR?
It seems to make the name less meaningful, and I'm not sure we gain from that.

Given the only setting that's

Reply:
Sorry, I missed your comment where you explained this.

I'm a strong believer in classes with a specific purpose.
`JwtAuthenticationTokenFactory` defines a purpose - it creates `JwtAuthenticationToken` objects. We can improve that name if we like, but I think having a class that does something specific is good.
`JwtRealms` doesn't really have a purpose. It's just some stuff relating to `JwtRealm` objects. And sometimes we need those, but when we can avoid them, I think we should.

I also don't understand the need to split the settings out of the class. What problem did we solve by having `JwtRealms` and `JwtRealmsSettings`?

I'm not especially attached to the original code, or the design decisions I made on the fly there, but it feels like a bunch of work has been done since then to rename/split classes (& add the registry of JWT realms) and the reasoning isn't clear to me.

Reply:
I was following the Realm vs Realms pattern, which differentiates via an `s`.
- Realm.java: `An authentication mechanism to which the default authentication org.elasticsearch.xpack.security.authc.AuthenticationService delegates the authentication process.`
- Realms.java: `Serves as a realms registry (also responsible for ordering the realms appropriately)`

I was leaning towards JwtRealms being a registry of JwtRealm instances. The idea was JwtRealm.token() hands off to JwtRealms.token(), and JwtRealms.token() could iterate all of the JwtRealm instances. It could inspect principal_claims.name for each JwtRealm instance, and use them instead of having the new list setting being decoupled from individual realm settings.

There is not enough time to dig down into that idea, so I removed the registry of JwtRealm instances. As for the name, I can leave it as JwtRealms.java for now, or append something to the name to help clarify it. I am open to suggestions. I don't think it will be specific to token, which is why I was looking for a different name than JwtAuthenticationTokenFactory.

Reply:
Renamed JwtRealms to JwtRealmsService. Added some extra comments.